runtime/src/tiered_storage.rs

```diff
@@ -126,14 +181,73 @@ mod tests {
             );
 
         let result = tiered_storage.write_accounts(&storable_accounts, 0);
-        // Expect the result to be TieredStorageError::Unsupported as the feature
-        // is not yet fully supported, but we can still check its partial results
-        // in the test.
-        assert!(result.is_err());
+
+        match (&result, &expected_result) {
+            (
+                Err(TieredStorageError::AttemptToUpdateReadOnly(_)),
+                Err(TieredStorageError::AttemptToUpdateReadOnly(_)),
+            ) => {}
+            (Err(TieredStorageError::Unsupported()), Err(TieredStorageError::Unsupported())) => {}
+            // we don't expect error type mis-match or other error types here
+            _ => {
+                assert_eq!(format!("{:?}", result), format!("{:?}", expected_result));
+            }
+        };
+
+        assert!(tiered_storage.is_read_only());
         assert_eq!(
             tiered_storage.file_size().unwrap() as usize,
             std::mem::size_of::<TieredStorageFooter>()
                 + std::mem::size_of::<TieredStorageMagicNumber>()
         );
     }
+
+    #[test]
+    fn test_new_meta_file_only() {
+        let tiered_storage_path = NamedTempFile::new().unwrap().path().to_path_buf();
```

Comment:
We actually don't because the file will be created when we created a TieredStorageFile:

```
     pub fn new_writable(file_path: impl AsRef<Path>) -> Result<Self, std::io::Error> {
         Ok(Self(
             OpenOptions::new()
                 .create_new(true)
                 .write(true)
                 .open(file_path)?,
         ))
     }
```

If the file exists, I think the `create_new(true)` will complains that the file already exists.